Shared/Supporting Files/Views/FavoritesView.swift

```diff
@@ -0,0 +1,153 @@
+                } label: {
+                    Image(systemName: "plus")
+                }
+                .sheet(isPresented: $isShowingSheet) {
+                    AddFavoriteView(samples: samples)
+                }
+            }
+        }
+    }
+}
+
+private extension FavoritesView {
+    /// A view to add a favorite sample from a searchable list.
+    struct AddFavoriteView: View {
+        /// All samples retrieved from the Samples directory.
+        let samples: [Sample]
+        
+        /// The action to dismiss the sheet.
+        @Environment(\.dismiss) private var dismiss: DismissAction
+        
+        /// The names of the favorited samples loaded from user defaults.
+        @AppStorage(UserDefaults.favoritedSamplesKey) private var favoritedNames: [String] = []
+        
+        /// The search query in the search bar.
+        @State private var query = ""
+        
+        /// The samples displayed in the search list.
+        var displayedSamples: [Sample] {
+            if query.isEmpty {
+                return samples
+            } else {
+                return samples.filter { $0.name.contains(query) }
+            }
+        }
+        
+        var body: some View {
+            NavigationView {
+                List {
+                    ForEach(displayedSamples, id: \.name) { sample in
+                        Button {
+                            if !favoritedNames.contains(sample.name) {
+                                favoritedNames.append(sample.name)
+                            }
+                            dismiss()
+                        } label: {
+                            HStack {
+                                Text(sample.name)
+                                Spacer()
+                            }
+                            .contentShape(Rectangle())
+                        }
+                        .buttonStyle(.plain)
+                    }
+                }
+                .listStyle(.inset)
+                .navigationBarTitleDisplayMode(.inline)
+                .toolbar {
+                    ToolbarItem(placement: .principal) {
+                        Text("Choose a sample to add to Favorites")
+                            .font(.subheadline)
+                    }
+                    ToolbarItem(placement: .navigationBarTrailing) {
+                        Button("Done") {
+                            dismiss()
+                        }
+                    }
+                }
+            }
+            .searchable(text: $query, placement: .navigationBarDrawer(displayMode: .always))
+        }
+    }
+}
+
+// MARK: User Defaults
+
+extension UserDefaults {
+    /// The key to read and write favorited sample names to the user defaults.
+    static var favoritedSamplesKey: String {
+        "favoritedSamples"
+    }
+}
+
+/// An extension allowing an array to be used with the app storage property wrapper.
+extension Array: RawRepresentable where Element: Codable {
```

Comment:
```suggestion
extension Array: RawRepresentable where Element == String {
```
Since we are only using this extension here, let's limit it to the very specific typing requirements so it doesn't cause too much a side effect.